doc/source/serve/architecture.md

```diff
@@ -49,7 +49,7 @@ Each replica maintains a queue of requests and executes requests one at a time,
 using `asyncio` to process them concurrently. If the handler (the deployment function or the `__call__` method of the deployment class) is declared with `async def`, the replica will not wait for the
 handler to run.  Otherwise, the replica will block until the handler returns.
 
-When making a request via [ServeHandle](serve-handle-explainer) instead of HTTP, the request is placed on a queue in the ServeHandle, and we skip to step 3 above.
+When making a request via a [DeploymentHandle](deployment-handle-explainer) instead of HTTP, the request is placed on a queue in the `DeploymentHandle`, and we skip to step 3 above.
```

Comment:
What do you think about linking to the DeploymentHandle section in Key Concepts instead of the DeploymentHandle section in Model Composition? I feel like if a reader comes across this and clicks into the link, they will be more lost if they get redirected to a specific guide like model composition (which is not always relevant for handles), but less lost if they get redirect to the key concepts page.

Reply:
great idea, i'll do that!

Reply:
I actually updated it to link to both (the key concept and the model composition page). PTAL